Arelle/Arelle
# Update EFM conformance suite scripts

#### Reason for change
- Scripts to run the EFM conformance suite needed improving
- EFM conformance suite needs to be consumable by future CI tests

#### Description of change
- Updated EFM bash scripts to be a simple python script.
- Added a CI test for the EFM conformance suite
- Fixed an issue in PackageManager where packagesConfig wasnt being initialized correctly after a close/reset call
- Fixed an issue in PluginManager where pluginConfig wasnt being initialized correctly after a close/reset call
- Fixed an issue in PluginManager where pluginConfig wasnt being reset correctly after a close call
- Added unit tests around the above fix for PackageManager
- Added unit tests around the above fixes for PluginManager

#### Steps to Test
- Run the CI tests with `pytest --disable-warnings tests/integration_tests`. Not all tests will pass, 24 should fail. Run will take about 3 hours.
- Run the testing script with `python -m scripts.conformance_suites.run_efm_conformance_suite`. Run will take about 2 hours and 45 minutes.

**review**:
@Arelle/arelle
